client/js/connection.js

```diff
@@ -81,6 +86,13 @@ function log(str) {
   toServer('trace', str);
 }
 
+setInterval(function() {
+  if(!pongReceived)
+    startWebSocket();
+  pongReceived = false;
+  toServer('ping');
```

Comment:
If this interval function runs before you are connected it will set pongReceived to false, not send the ping, and the next time it runs it will detect that it didn't receive a pong and call startWebSocket which will open a second connection since the previous connection isn't closed.

Instead, you should probably install this interval after setting up the connection or at the very least do nothing if not currently connected.